### PR TITLE
feat: [DEL-5826]: Decrypt secrets in init container

### PR DESCRIPTION
**Pull request by mdmilic**
## Describe your changes
With this change K8S Task Runner will create init container for decrypting secrets (if needed). This secret decryption container will decrypt secrets and store results in a shared volume so main task container can use them. 

Testing:
* Tested simple tasks with secret vars. Confirmed they get decrypted and stored in output location.
* Works for Vault right now. 

I tried GCP KMS, but the task container didn't have sufficient permissions to read them (the same is most likely true for some other providers as well). Need to work with someone from platform team to find out the exact appropriate permissions per secret provider. However this can be done as a separate change and should not block this PR (this is only used for bijou delegate and internal demos)

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43092)

### CODE REVIEW

File: 931-delegate-decryption-service/src/java/io/harness/decryption/delegate/SecretsHandler.java [hunk 1]
```diff
@@ -7,12 +7,58 @@
...
+import lombok.Getter;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 
 @RequiredArgsConstructor(onConstructor_ = @Inject)
+@Getter
 @Slf4j
 public class SecretsHandler {
-  public static void main(String[] args) {}
+  private static final String SECRET_INPUT_LOCATION = "/opt/harness/secrets/input/secrets.bin";
```
Comment (gn-harness) [inline 931-delegate-decryption-service/src/java/io/harness/decryption/delegate/SecretsHandler.java:33]: There would be multiple tasks running at any point in time - this path is specific to the container inside which task runs? and delegate would mount it on the container before executing it?

File: 260-delegate/src/main/java/io/harness/delegate/service/core/k8s/K8STaskRunner.java [hunk 2]
```diff
@@ -88,25 +117,34 @@ private void createTaskJob(final String taskId, final V1Volume taskPackageVolume
...
-                        .addVolume(delegateConfigVolume, "/etc/delegate-config")
-                        .addVolume(secretVolume, "/etc/secret")
-                        .addEnvVar("ACCOUNT_ID", delegateConfiguration.getAccountId())
-                        .addEnvVar("TASK_ID", taskId)
-                        .addEnvVar("DELEGATE_NAME", "");
+                        .namespace(HARNESS_DELEGATE_NG);
+
+    if (secretVolume.isPresent()) {
+      job.addInitContainer("secret-decryption", "us.gcr.io/gcr-play/secret-provider:secrets")
+          .addVolume(secretVolume.get(), "/opt/harness/secrets/input/")
```
Comment (gn-harness) [inline 260-delegate/src/main/java/io/harness/delegate/service/core/k8s/K8STaskRunner.java:124]: use shared constants

**Review by gn-harness — APPROVED** [review]

**Review by gn-harness — COMMENTED** [review]
> Reviewed 16 of 32 files at r1, 7 of 7 files at r2, 9 of 9 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43092)* status: :shipit: complete! ✓

**Review by xingchi-jin — DISMISSED** [review]
> Reviewed 16 of 32 files at r1, 1 of 7 files at r2, 6 of 9 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43092)* status: :shipit: complete! ✓

**Review by mdmilic — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43092)* status: 2. unresolved discussions. 

___
*[`260-delegate/src/main/java/io/harness/delegate/service/core/k8s/K8STaskRunner.java` line 124 at r3](https://reviewable.io/reviews/harness/harness-core/43092#-NMfMg1AvkEoJaMxqh-Y-r3-124:-NMp6lqs1IUSDAI7hiBO:b10aqkr) ([raw file](https://github.com/harness/harness-core/blob/012a27b75da9014f37fb5353f13f18a1311f38d0/260-delegate/src/main/java/io/harness/delegate/service/core/k8s/K8STaskRunner.java#L124)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

use shared constants
</blockquote></details>

Will try merge like this and then make a change for it to avoid getting all approvals again. 
___
*[`931-delegate-decryption-service/src/java/io/harness/decryption/delegate/SecretsHandler.java` line 33 at r3](https://reviewable.io/reviews/harness/harness-core/43092#-NMfMg1AvkEoJaMxqh-d-r3-33:-NMp6vay5ivU_V2ikb2a:b-jhi5st) ([raw file](https://github.com/harness/harness-core/blob/012a27b75da9014f37fb5353f13f18a1311f38d0/931-delegate-decryption-service/src/java/io/harness/decryption/delegate/SecretsHandler.java#L33)):*
<details><summary><i>Previously, gn-harness (Gaurav Nanda) wrote…</i></summary><blockquote>

There would be multiple tasks running at any point in time - this path is specific to the container inside which task runs? and delegate would mount it on the container before executing it?
</blockquote></details>

Correct, this is the path inside container. Would be the same in each task container.

**Review by gn-harness — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43092)* status: 2. unresolved discussions. 0 of 1 approvals obtained.

**Review by xingchi-jin — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43092)* status: 2. unresolved discussions.

**Review by gn-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43092)* status: :shipit: complete! ✓

**Review by raghuAtWings — APPROVED** [review]
> Reviewed 16 of 32 files at r1, 7 of 7 files at r2, 8 of 9 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43092)* status: :shipit: complete! ✓

**Pull request merged**